File: ai_tnhn/ai-api-tnhn/router/organization.go
package router

import (
	"ai-api-tnhn/handler"
	"ai-api-tnhn/router/middleware"

	"github.com/gin-gonic/gin"
)

func (h *HandlerFuncs) OrganizationRoutes(g *gin.RouterGroup, mid middleware.Middleware, orgHandler *handler.OrganizationHandler) {
	org := g.Group("/organizations", mid.MidBasicType()) // Apply Authentication Middleware
	{
		org.POST("", orgHandler.Create)
		org.PUT("/:id", orgHandler.Update)
		org.DELETE("/:id", orgHandler.Delete)
		org.GET("/:id", orgHandler.GetByID)
		org.GET("", orgHandler.List)
	}
}

File: ai_tnhn/ai-frontend-tnhn/src/api/organization.js
import axiosClient from './axiosClient';

const organizationApi = {
    getAll: (params) => {
        return axiosClient.get('/admin/organizations', { params });
    },
    getSelectionList: () => {
        return axiosClient.get('/admin/organizations/selection');
    },
    getById: (id) => {
        return axiosClient.get(`/admin/organizations/${id}`);
    },
    create: (data) => {
        return axiosClient.post('/admin/organizations', data);
    },
    update: (id, data) => {
        return axiosClient.put(`/admin/organizations/${id}`, data);
    },
    delete: (id) => {
        return axiosClient.delete(`/admin/organizations/${id}`);
    }
};

export default organizationApi;
